python/cudf/cudf/utils/ioutils.py

```diff
@@ -1154,6 +1158,10 @@ def get_filepath_or_buffer(
             # path_or_data need not be a filepath like string
             if os.path.exists(paths[0]):
                 path_or_data = paths if len(paths) > 1 else paths[0]
+        
+        elif _is_s3_filesystem(fs):
+            fs = pyarrow.fs.S3FileSystem()
+            path_or_data = fs.open_input_file(paths[0]) 
```

Comment:
This logic might need to be moved to `csv.py` for now, to ensure this codepath is only followed for `read_csv` calls. In the future, this could be implemented here when other readers support this option as well